### PR TITLE
Rust: Restrict variables to lowercase (for now).

### PR DESCRIPTION
**Pull request by geoffw0**
Temporary workaround for misrecognising enum constants as variable names.  This is not always accurate (see the comment) but should lean towards too few things being recognised as variables rather than too many, which is better for the quality of results.

@hvitved

### CODE REVIEW

File: rust/ql/lib/codeql/rust/elements/internal/VariableImpl.qll [hunk 1]
```diff
@@ -85,7 +85,13 @@ module Impl {
     private AstNode definingNode;
     private string name;
 
-    Variable() { this = MkVariable(definingNode, name) }
+    Variable() {
+      this = MkVariable(definingNode, name) and
+      // exclude for now anything starting with an uppercase character, which may be an enum constant (e.g. `None`). This excludes
+      // static and constant variables (UPPERCASE), which we don't appear to recognize yet anyway. This also assumes programmers
+      // follow the naming guidelines, which they generally do, but they're not enforced.
+      not name.charAt(0).isUppercase()
```
Comment (hvitved) [inline rust/ql/lib/codeql/rust/elements/internal/VariableImpl.qll:93]: I think it's better to put this restriction in the `variableDecl` predicate.

Comment (geoffw0) [reply rust/ql/lib/codeql/rust/elements/internal/VariableImpl.qll:93]: I'll try that.  It didn't have an effect in `VariableAccessCand`, which is where I originally thought it should go.

Comment (geoffw0) [reply rust/ql/lib/codeql/rust/elements/internal/VariableImpl.qll]: Updated.

**Review by hvitved — DISMISSED** [review]

**Comment by geoffw0** [comment]
There was a test failure in one of the library tests (also a `match` case of `None`).

**Review by hvitved — APPROVED** [review]

**Pull request merged**